core/domain/takeout_service_test.py

```diff
@@ -48,10 +48,388 @@
+            last_edited_an_exploration=self.GENERIC_DATE,
+            profile_picture_data_url=self.GENERIC_IMAGE_URL,
+            default_dashboard='learner', creator_dashboard_display_pref='card',
+            user_bio=self.GENERIC_USER_BIO,
+            subject_interests=self.GENERIC_SUBJECT_INTERESTS,
+            first_contribution_msec=1,
+            preferred_language_codes=self.GENERIC_LANGUAGE_CODES,
+            preferred_site_language_code=self.GENERIC_LANGUAGE_CODES[0],
+            preferred_audio_language_code=self.GENERIC_LANGUAGE_CODES[0],
+            display_alias=self.GENERIC_DISPLAY_ALIAS
+        ).put()
+        user_models.UserSettingsModel(
+            id=self.PROFILE_ID_1,
+            gae_id=self.USER_GAE_ID_1,
+            email=self.USER_1_EMAIL,
+            role=self.PROFILE_1_ROLE,
+            username=None,
+            normalized_username=None,
+            last_agreed_to_terms=self.GENERIC_DATE,
+            last_started_state_editor_tutorial=None,
+            last_started_state_translation_tutorial=None,
+            last_logged_in=self.GENERIC_DATE,
+            last_created_an_exploration=None,
+            last_edited_an_exploration=None,
+            profile_picture_data_url=None,
+            default_dashboard='learner', creator_dashboard_display_pref='card',
+            user_bio=self.GENERIC_USER_BIO,
+            subject_interests=self.GENERIC_SUBJECT_INTERESTS,
+            first_contribution_msec=None,
+            preferred_language_codes=self.GENERIC_LANGUAGE_CODES,
+            preferred_site_language_code=self.GENERIC_LANGUAGE_CODES[0],
+            preferred_audio_language_code=self.GENERIC_LANGUAGE_CODES[0],
+            display_alias=self.GENERIC_DISPLAY_ALIAS_2
+        ).put()
+
+    def set_up_trivial(self):
+        """Setup for trivial test of export_data functionality."""
+        super(TakeoutServiceProfileUserUnitTests, self).setUp()
+        user_models.UserSettingsModel(
+            id=self.USER_ID_1,
+            gae_id=self.USER_GAE_ID_1,
+            email=self.USER_1_EMAIL,
+            role=self.USER_1_ROLE
+        ).put()
+        user_models.UserSettingsModel(
+            id=self.PROFILE_ID_1,
+            gae_id=self.USER_GAE_ID_1,
+            email=self.USER_1_EMAIL,
+            role=self.PROFILE_1_ROLE
+        ).put()
+        user_models.UserSubscriptionsModel(id=self.USER_ID_1).put()
+        user_models.UserSubscriptionsModel(id=self.PROFILE_ID_1).put()
+
+    def test_export_nonexistent_user_raises_error(self):
+        """Setup for nonexistent user test of export_data functionality."""
+        with self.assertRaisesRegexp(
+            user_models.UserSettingsModel.EntityNotFoundError,
+            'Entity for class UserSettingsModel with id fake_user_id '
+            'not found'):
+            takeout_service.export_data_for_user('fake_user_id')
+
+    def test_export_data_for_user_trivial_is_correct(self):
```

Comment:
I thought, for profile users, we are just not supporting this functionality (per previous discussion in the review comments)? If so, the handler should return 404 or similar.

Reply:
I thought what you meant [here](https://github.com/oppia/oppia/pull/10241#discussion_r469751780) was that when we provide a `user_id` for takeout, we only takeout that user's data (and not any of its linked profiles if it's a full user).
Why to completely disallow takeout for profiles? 
Also here the test which you gave as an example only checks whether takeout for that profile user only is done properly. Was the name misleading or did I understand anything incorrectly?

Reply:
See the comment in parentheses here: https://github.com/oppia/oppia/pull/10241#discussion_r469751780. If we haven't figured out how auth for profiles works then we shouldn't support profile export yet. You can have the method raise NotImplementedError for now when called for profile users, and add a TODO. We really do not want the wrong person to get the data, so this is safer for now.

(As a side note, I suggest adding "profile" or "full" into your test names, to help distinguish the type of user that is under consideration.)

Reply:
@seanlip Sorry missed it earlier. Done now. PTAL! A `NotImplementedError` raised in the main function(`export_data_for_user`) and in the tests I have populated the required models for Profile users, but checked that calling the `export_data_for_user` method raises error.